src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl

```diff
@@ -1564,49 +1564,67 @@ KERNEL(sdpa_opt)(
     #endif
 #endif
 #endif
-
                     MAKE_VECTOR_TYPE(OUTPUT_TYPE, TARGET_SEQ_LEN_BLOCK_SIZE) qk_val;
                     unroll_for (uint seq_idx = 0; seq_idx < TARGET_SEQ_LEN_BLOCK_SIZE; seq_idx++) {
                         qk_val[seq_idx] = slm_qk_vals[seq_idx][seq_len + sglid];
                     }
-
 #if IS_KV_COMPRESSED
                     const uint comp_offset = GET_COMPRESSION_INDEX(VALUE_COMPRESSION_SCALE, b_idx, b1_idx / BROADCAST_GROUP_SIZE, start_partition_idx + seq_len + sglid, 0);
                     VALUE_COMPRESSION_SCALE_TYPE comp_scale = val_scale[comp_offset];
 #if USE_ASYMMETRIC_QUANTIZATION
                     VALUE_COMPRESSION_SCALE_TYPE comp_zp = val_scale[comp_offset + 1];
 #endif
 #endif
+                    #ifdef V_HEAD_SIZE_LEFTOVER
+                    // splitting to two cases for supressing reg spill : block_read & eltwise read 
+                    if (head_size_idx < V_HEAD_SIZE && sgid < SUBGROUPS_PER_WG - 1) {
```

Comment:
nit:
Should single condition be enough?
```suggestion
                    if (sgid < SUBGROUPS_PER_WG - 1) {
```

Reply:
yeah it was redundant. I removed!